src/circuit/tx.rs

```diff
@@ -175,14 +187,52 @@ pub fn c_transfer<C:CS, P:PoolParams<Fr=C::Fr>>(
         (&s.tx.input.1[i].p_d - c_derive_key_p_d(&s.tx.input.1[i].d.as_num(), &eta_bits, params).x).assert_zero();
     }
 
+    // Check daily turnover limit and transfer limit
+    let in_account = &s.tx.input.0;
+    let out_account = &s.tx.output.0;
+    {
+        let is_new_day = c_comp(&p.day.as_num(), &in_account.last_action_day.as_num(), DAY_SIZE_BITS);
+
+        let delta_value_is_positive = c_comp(&total_value, &p.derive_const(&Num::ZERO), TURNOVER_SIZE_BITS);
+        let delta_value_abs = total_value.clone().switch(&delta_value_is_positive, &-&total_value);
+        let tx_turnover = delta_value_abs + &out_notes_sum;
+        let turnover = tx_turnover.switch(&is_new_day, &(in_account.daily_turnover.as_num() + &tx_turnover));
+
+        // Check transfer limit
+        c_comp(&out_notes_sum, &p.transfer_limit.as_num(), BALANCE_SIZE_BITS).assert_const(&false);
+        // Check that current day >= last_action_day
+        (&is_new_day | p.day.is_eq(&in_account.last_action_day)).assert_const(&true);

```

Comment:
Just curios, will
```c_comp(&in_account.last_action_day.as_num(), &p.day.as_num(), DAY_SIZE_BITS).assert_const(&false);```
be more optimal?

Reply:
It seems that we need one is_eq and one c_comp anyway. That constraint will be more optimal, but to compute turnover we need to know if p.day is greater or if p.day is equal to in_account.last_action_day. So it seems that we just move is_eq to some other place.